distribution/push_v2.go

```diff
@@ -188,60 +184,24 @@ func (p *pusher) pushTag(ctx context.Context, ref reference.NamedTagged, id dige
 
 	putOptions := []distribution.ManifestServiceOption{distribution.WithTag(ref.Tag())}
 	if _, err = manSvc.Put(ctx, manifest, putOptions...); err != nil {
-		if runtime.GOOS == "windows" {
-			log.G(ctx).Warnf("failed to upload schema2 manifest: %v", err)
-			return err
-		}
-
-		// This is a temporary environment variables used in CI to allow pushing
-		// manifest v2 schema 1 images to test-registries used for testing *pulling*
-		// these images.
-		if os.Getenv("DOCKER_ALLOW_SCHEMA1_PUSH_DONOTUSE") == "" {
-			if err.Error() == "tag invalid" {
-				msg := "[DEPRECATED] support for pushing manifest v2 schema1 images has been removed. More information at https://docs.docker.com/registry/spec/deprecated-schema-v1/"
-				log.G(ctx).WithError(err).Error(msg)
-				return errors.Wrap(err, msg)
-			}
-			return err
+		if err.Error() == "tag invalid" {
```

Comment:
Hmm, where is this error defined?

Reply:
Good question; ISTR this was a bit of a hack, because there was no other way to detect this from the request, so this may have been a "best effort" match for what the registry returned.

Reply:
Right, but I don't think that distribution ever returned this error string.

https://grep.app/search?f.lang=Go&q=%22tag+invalid%22 points only to some `shellhub` using this string.

Wondering if we could just drop it.

Reply:
LOL; looks like I found the best reference in the PR that disabled push... 5 Years ago; I think it was Docker Hub that returned it as error, so it may have been docker-hub specific; https://github.com/moby/moby/pull/41295#discussion_r464506385

All that said, I don't think this PR make the situation worse; it "may" work and show the more useful error, otherwise just returns the original error as-is.

Honestly, not even sure if we can hit this code at all now; hitting this code would mean that we _somehow_ managed to get a v1 image into the daemon? So after this PR makes its way in, I think we should look at removing all the special handling here (at most "unknown / unsupported manifest type").

Reply:
Sure, we can handle that in a follow up